lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -753,6 +809,44 @@ func cgo_handle_fastpath_pdu(handle C.uintptr_t, data *C.uint8_t, length C.uint3
 	return client.handleRDPFastPathPDU(goData)
 }
 
+func (c *Client) readRDPLicense(majorVersion, minorVersion uint16, issuer, company, productID string) ([]byte, error) {
+	log := c.cfg.Logger.With(
+		"issuer", issuer,
+		"company", company,
+		"version", fmt.Sprintf("%d.%d", majorVersion, minorVersion),
+		"product", productID,
+	)
+
+	license, err := c.cfg.LicenseStore.ReadRDPLicense(context.Background(), majorVersion, minorVersion, issuer, company, productID)
+	switch {
+	case trace.IsNotFound(err):
+		log.InfoContext(context.Background(), "existing RDP license not found")
+	case err != nil:
+		log.ErrorContext(context.Background(), "could not look up existing RDP license", "error", err)
+	case len(license) > 0:
+		log.InfoContext(context.Background(), "found existing RDP license")
```

Comment:
what should happen in case of `err == nil && len(license) == 0`?

Reply:
That's not something I've commonly seen in Go code. It's generally safe to assume that a nil error means we're returning valid data.

Reply:
So why use additional case here? can we skip it and simply log after switch?

Reply:
I'm not sure I follow?

I want to be able to tell from the logs if:
1. We know there wasn't an existing license.
2. We know there was an existing license.
3. We don't know if there was a license or not because we encountered an error.

Reply:
Right now the switch is not exhaustive, case where `err == nil` and `len(license) == 0` will log nothing.
So, we either don't care about the length (we assume it's `>0` when `err==nil`) and we can skip the length check (by either returning early in case of errors above and logging happy path outside of switch or changing last case to default) or we do care and should log something different in that case

Reply:
Fair enough, we can change the last case to a default case instead.

Reply:
We could error upon trying to write an empty license (if that's ever plausible) and treat an empty license as a not found error, too.

For what it's worth we seem to always avoid writing items with a blank value in the backend - nothing bad should happen, but...